# Student Performance Dataset Analysis

# Aim

The aim of this project is to analyze student performance data, identify factors affecting academic performance, perform data cleaning, and generate meaningful insights using exploratory data analysis.

# Dataset Description

This dataset contains student demographic, academic, and behavioral information. The dataset is used to study factors that influence student GPA and overall academic performance.

In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [25]:
df = pd.read_csv("Student_performance_data _.csv")

In [26]:
df.head()

,StudentID,Age,Gender,Ethnicity,ParentalEducation,StudyTimeWeekly,Absences,Tutoring,ParentalSupport,Extracurricular,Sports,Music,Volunteering,GPA,GradeClass
0,1001,17,1,0,2,19.833723,7,1,2,0,0,1,0,2.929196,2.0
1,1002,18,0,0,1,15.408756,0,0,1,0,0,0,0,3.042915,1.0
2,1003,15,0,2,3,4.210570,26,0,2,0,0,0,0,0.112602,4.0
3,1004,17,1,0,3,10.028829,14,0,3,1,0,0,0,2.054218,3.0
4,1005,17,1,0,2,4.672495,17,1,3,0,0,0,0,1.288061,4.0


In [27]:
df.tail()

,StudentID,Age,Gender,Ethnicity,ParentalEducation,StudyTimeWeekly,Absences,Tutoring,ParentalSupport,Extracurricular,Sports,Music,Volunteering,GPA,GradeClass
2387,3388,18,1,0,3,10.680555,2,0,4,1,0,0,0,3.455509,0.0
2388,3389,17,0,0,1,7.583217,4,1,4,0,1,0,0,3.279150,4.0
2389,3390,16,1,0,2,6.805500,20,0,2,0,0,0,1,1.142333,2.0
2390,3391,16,1,1,0,12.416653,17,0,2,0,1,1,0,1.803297,1.0
2391,3392,16,1,0,2,17.819907,13,0,2,0,0,0,1,2.140014,1.0


The first and last records were inspected successfully.

In [28]:
df.shape

(2392, 15)

The dataset contains 2392 rows and 15 columns.

# Dataset Summary

Number of Records : 2392

Number of Features : 15

Target Variable : GPA

Dataset Type : Student Performance Dataset

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2392 entries, 0 to 2391
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   StudentID          2392 non-null   int64  
 1   Age                2392 non-null   int64  
 2   Gender             2392 non-null   int64  
 3   Ethnicity          2392 non-null   int64  
 4   ParentalEducation  2392 non-null   int64  
 5   StudyTimeWeekly    2392 non-null   float64
 6   Absences           2392 non-null   int64  
 7   Tutoring           2392 non-null   int64  
 8   ParentalSupport    2392 non-null   int64  
 9   Extracurricular    2392 non-null   int64  
 10  Sports             2392 non-null   int64  
 11  Music              2392 non-null   int64  
 12  Volunteering       2392 non-null   int64  
 13  GPA                2392 non-null   float64
 14  GradeClass         2392 non-null   float64
dtypes: float64(3), int64(12)
memory usage: 280.4 KB


The dataset contains numerical and categorical features. Data types were inspected to identify possible corrections.

| Feature         | Type            | Description          |
| --------------- | --------------- | -------------------- |
| StudentID       | Identifier      | Unique Student ID    |
| Age             | Numerical       | Student Age          |
| Gender          | Categorical     | Student Gender       |
| Ethnicity       | Categorical     | Student Ethnicity    |
| StudyTimeWeekly | Numerical       | Weekly Study Hours   |
| Absences        | Numerical       | Number of Absences   |
| Tutoring        | Categorical     | Tutoring Support     |
| GPA             | Target Variable | Academic Performance |


GPA is selected as the main performance indicator.
StudyTimeWeekly, Absences, Tutoring, and ParentalSupport may influence GPA.

In [30]:
df.isnull().sum()

,0
StudentID,0
Age,0
Gender,0
Ethnicity,0
ParentalEducation,0
StudyTimeWeekly,0
Absences,0
Tutoring,0
ParentalSupport,0
Extracurricular,0


No missing values were found in the dataset.

In [31]:
df.duplicated().sum()

np.int64(0)

Duplicate records were checked to ensure data consistency.

In [32]:
df = df.drop_duplicates()

The dataset was checked for duplicates, missing values, and inconsistent labels. No major quality issues were found.

In [33]:
df.to_csv("cleaned_student_performance.csv", index=False)

A cleaned version of the dataset was saved separately.

In [34]:
df.describe()

,StudentID,Age,Gender,Ethnicity,ParentalEducation,StudyTimeWeekly,Absences,Tutoring,ParentalSupport,Extracurricular,Sports,Music,Volunteering,GPA,GradeClass
count,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000,2392.000000
mean,2196.500000,16.468645,0.510870,0.877508,1.746237,9.771992,14.541388,0.301421,2.122074,0.383361,0.303512,0.196906,0.157191,1.906186,2.983696
std,690.655244,1.123798,0.499986,1.028476,1.000411,5.652774,8.467417,0.458971,1.122813,0.486307,0.459870,0.397744,0.364057,0.915156,1.233908
min,1001.000000,15.000000,0.000000,0.000000,0.000000,0.001057,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,1598.750000,15.000000,0.000000,0.000000,1.000000,5.043079,7.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000,1.174803,2.000000
50%,2196.500000,16.000000,1.000000,0.000000,2.000000,9.705363,15.000000,0.000000,2.000000,0.000000,0.000000,0.000000,0.000000,1.893393,4.000000
75%,2794.250000,17.000000,1.000000,2.000000,2.000000,14.408410,22.000000,1.000000,3.000000,1.000000,1.000000,0.000000,0.000000,2.622216,4.000000
max,3392.000000,18.000000,1.000000,3.000000,4.000000,19.978094,29.000000,1.000000,4.000000,1.000000,1.000000,1.000000,1.000000,4.000000,4.000000


Descriptive statistics were generated to understand the distribution of numerical variables.

# Observations

1. The dataset contains 2392 student records and 15 features.

2. No missing values were found, indicating good data quality.

3. GPA is the primary indicator of academic performance.

4. Students with higher weekly study time generally tend to achieve better GPA scores.

5. Higher absences may negatively affect academic performance.

6. Tutoring support may contribute positively to student outcomes.

7. Parental support appears to influence student performance.

8. Academic performance depends on multiple educational and behavioral factors.

# Conclusion

The dataset was successfully analyzed and found to be clean and well-structured. Study time, attendance, tutoring, and parental support appear to influence GPA. The dataset is suitable for further analysis and machine learning applications.